src/interfaces/plugins.ts

```diff
@@ -679,17 +696,30 @@ module.exports = (theApp: any) => {
     })
 
     if (typeof plugin.registerWithRouter == 'function') {
-      plugin.registerWithRouter(router)
```

Comment:
The `registerWithRouter` check is left here dangling, now guarding just `getOpenApi`. I think these should be moved inside `doRegisterWithRouter` also.